xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/documentTags.vm

```diff
@@ -109,31 +109,27 @@
 #else ## display
     $xwiki.ssfx.use('uicomponents/viewers/tags.css', {'forceSkinAction': true, 'colorTheme': "$!{themeDocFullName}"})##
     $xwiki.jsfx.use('uicomponents/viewers/tags.js', true)##
-<div class="doc-tags" id="${tagsId}">
+
     #if($xwiki.tag)
         #set($hasTagsPlugin = true)
         #set($tags = $xwiki.tag.getTagsFromDocument($doc.fullName))
     #else
         #set($hasTagsPlugin = false)
         #set($tags = $doc.getTagList())
     #end
-    ##   #if($!tags.size() == 0)
-    ##     No tags
-    ##   #else
-    $services.localization.render('core.tags.list.label')
-    #foreach($tag in $tags)
-        #displayTag($tag)
-    #end
-    ##   #end
-    #if($hasedit)
-        #if($xwiki.tag)
-          <div class="tag-tool tag-add">#if("$!{request.showTagAddForm}" == '')<a href="$doc.getURL('view', "showTagAddForm=true&amp;$!{escapetool.url(${request.queryString})}#${tagsId}")" title="$services.localization.render('core.tags.add.tooltip')" rel="nofollow">[+]</a>#else #displayAddForm()#end</div>
-        #else
-            ## TODO
+    <div class="doc-tags #if(!$hasedit && $tags.isEmpty() )hidden#end" id="${tagsId}">
+        $services.localization.render('core.tags.list.label')
+        #if($hasedit || !$tags.isEmpty())
+            #foreach($tag in $tags)
+                #displayTag($tag)
+            #end
+            #if($hasedit && $xwiki.tag)
+                <div class="tag-tool tag-add">#if("$!{request.showTagAddForm}" == '')<a href="$doc.getURL('view', "showTagAddForm=true&amp;$!{escapetool.url(${request.queryString})}#${tagsId}")" title="$services.localization.render('core.tags.add.tooltip')" rel="nofollow">[+]</a>#else #displayAddForm()#end</div>
+            #end
```

Comment:
This is overly complicated.. AFAICS the only change required to the original source code is to wrap the ``$services.localization`` line inside an ``#if``. Basically to show the "Tags:" label only if there are tags or the user can add a new tag.

Reply:
Check

Reply:
Maybe I wasn't clear enough, but for me the diff should show only a **single** line as modified, and that line is the ``$services.localization`` line. I don't think you need other changes. I suggest you start over from the **original** source. Don't create a new PR, just read my comment above and apply my suggestion on the original source.